src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/FrameConnection.cs

```diff
@@ -242,10 +267,40 @@ public void Tick(DateTimeOffset now)
 
                 if (_timeoutAction == TimeoutAction.SendTimeoutResponse)
                 {
+                    SetTimeoutResponse();
+                }
+
+                Timeout();
+            }
+            else if (_timingReads)
+            {
+                _readTimingElapsed += timestamp - _lastTimestamp;
+
+                if (_readTimingElapsed > _frame.RequestBodyTimeout.Ticks)
```

Comment:
I'll accept as much data as you can send me in 60s? I don't see that as a realistic scenario.

A full request/response processing time limit seems more realistic. E.g. my SLA for this request is 1s, and if I can't fulfill it I need to cancel.